File: algorithms/easy_questions/two_sum/two_sum.java
import java.util.HashMap;
import java.util.Map;

public class two_sum {
    public static int[] brutFroce(int[] nums, int target) {
        // length of nums
        int length = nums.length;

        for (int i = 0; i < length - 1; i++) {
            for (int j = i + 1; j < length; j++) {
                if (nums[i] + nums[j] == target) {
                    return new int[]{i, j};
                }
            }
        }
        return new int[]{};
    }

    public static int[] complements(int[] nums, int target) {
        Map<Integer, Integer> numDict = new HashMap<>();
        int length = nums.length;

        for (int i = 0; i < length; i++) {
            int complement = target - nums[i];
            if (numDict.containsKey(complement)) {
                return new int[]{numDict.get(complement), i};
            }
            numDict.put(nums[i], i);
        }

        return new int[]{};
    }

    public static void main(String[] args) {
        int[] test1 = {5, 6, 12, 9, 13};
        int target1 = 14;
        int[] test2 = {1, 2, 3, 4, 5};
        int target2 = 7;

        int[] result1 = complements(test1, target1);
        int[] result2 = complements(test2, target2);

        for (int value : result1) {
            System.out.print(value + " ");
        }
        System.out.println();

        for (int value : result2) {
            System.out.print(value + " ");
        }
        System.out.println();
    }
}

File: challenges/algorithms/easy_challenges/two_sum/two_sum.py
class Solution:
    def bruteForce(nums: list[int], target: int) -> list[int]:
        # length of nums
        length = len(nums)
        # Loops for finding target
        for i in range(length - 1):
            for j in range(i + 1, length):
                if nums[i] + nums[j] == target:
                    return [i, j]
        return []
    
    def complements(nums: list[int], target: int) -> list[int]:
        # hash table
        numDict = {}
        # length of nums
        length = len(nums)
        
        for i in range(length):
            complement = target - nums[i]
            if complement in numDict:
                return [numDict[complement], i]
            numDict[nums[i]] = i

test1 = [5,6,12,9,13]
target1 = 14
test2 = [1,2,3,4,5]
target2 = 7

print (Solution.complements(test1, target1))
print (Solution.complements(test2, target2))